db/migrate/20240605134916_add_notes_and_diary_comments_counter_caches.rb

```diff
@@ -0,0 +1,6 @@
+class AddNotesAndDiaryCommentsCounterCaches < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :diary_comments_count, :integer, :default => 0
+    add_column :users, :note_comments_count, :integer, :default => 0
```

Comment:
These need to call `reset_counters` for any user that already has comments or the counts won't be correct - see `db/migrate/20121005195010_add_diary_entry_counter_caches.rb` for a way to to it reasonably efficiently.

There are less than 10k users with diary comments so that is probably fine but there are nearly half a million with not comments which may prove to be a problem in terms of time take to run the migration and populate the counters to we might have to do it with a rake task rather than in the migration.